packages/url-state-provider/src/v2/encode.spec.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Juno contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import encode from "./encode"
+import testCases from "./testCases"
+
+describe("encode", () => {
+  it.each(testCases)("[$id] should successfully encode given input", ({ decoded: input, encoded: output }) => {
+    expect(encode(input, { skipEmptyString: false, skipNull: false })).toBe(output)
+  })
+
+  it("should remove empty string and null by default", () => {
+    expect(
+      encode({
+        key: "value",
+        null: null,
+        empty: "",
+      })
+    ).toBe("key=value")
+  })
+
+  it.each`
```

Comment:
I would really try to avoid it :) 
usually it will make failed test less readable and quite hard to see what is going on, test must be explicit

Reply:
Ah I've exactly the opposite opinion. I find `it.each` much more readable because this way you can find different scenarios/test cases all at once and can add a new one without repeating too much of the code. For example imagine converting the following to each individual test. Also if you have good description for each scenario it is quite easier to find which one is failing, so this is not a problem of `it.each` rather how one describes the test cases.

<img width="823" alt="Screenshot 2025-01-14 at 14 02 56" src="https://github.com/user-attachments/assets/7f1a95cc-3c5c-43b8-a03a-23787567b783" />